Source/SDK/ClientBuilder.cs

```diff
@@ -238,6 +240,17 @@ public ClientBuilder WithLogging(ILoggerFactory factory)
             return this;
         }
 
+        /// <summary>
+        /// Sets the default <see cref="JsonSerializerSettings"/> .
+        /// </summary>
+        /// <param name="jsonSerializerSettings">The default <see cref="JsonSerializerSettings"/>.</param>
+        /// <returns>The client builder for continuation.</returns>
+        public ClientBuilder WithJsonSerializerSettings(JsonSerializerSettings jsonSerializerSettings)
+        {
+            _jsonSerializerSettings = jsonSerializerSettings;
+            return this;
+        }
```

Comment:
We discussed this internally and landed on that we'd want to provide additional functionality in the future for having custom settings for serializing embeddings and projections as well as events. Since this PR only touches on events we think that it's better to change the name of the method so that it reflects what it really does at the moment
```suggestion
        /// <summary>
        /// Sets the <see cref="JsonSerializerSettings"/> for serializing events.
        /// </summary>
        /// <param name="jsonSerializerSettings">The default <see cref="JsonSerializerSettings"/>.</param>
        /// <returns>The client builder for continuation.</returns>
        public ClientBuilder WithEventSerializerSettings(JsonSerializerSettings jsonSerializerSettings)
        {
            _jsonSerializerSettings = jsonSerializerSettings;
            return this;
        }
```